### PR TITLE
polkitagent, loopback: Port deprecated GSimpleAsyncResult to GTask

### PR DESCRIPTION
**Pull request by martinpitt**
GSimpleAsyncResult got deprecated in GLib 2.46 in favor of GTask
(available since 2.36).

This will enable us to bump our glib requirement to ≥ 2.46 in the future
(otherwise building fails because of -Wdeprecated).

### CODE REVIEW

**Comment by martinpitt** [comment]
This is not complete yet, I just want to test the cockpitpolkitagent.c changes as these aren't covered by unit tests.

**Comment by martinpitt** [comment]
I don't currently have time to deal with the other two. I have WIP patches for  [cockpitauth](https://paste.fedoraproject.org/paste/LkSjwwynyRGEI2P80rThyw) and [cockpitconnect](https://paste.fedoraproject.org/paste/i9b6oNDUCFwgQTfOPSyvtA), but neither of them works (unit tests fail already) -- it seems I'm missing something rather subtle there.

But these two ports should be fine already, so I'd like to land these before they bitrot.

File: src/common/cockpitloopback.c [hunk 1]
```diff
@@ -66,16 +66,15 @@ cockpit_loopback_next_async (GSocketAddressEnumerator *enumerator,
...
   GError *error = NULL;
 
   address = cockpit_loopback_next (enumerator, cancellable, &error);
   g_assert (error == NULL);
 
-  res = g_simple_async_result_new (G_OBJECT (self), callback, user_data, NULL);
-  g_simple_async_result_set_op_res_gpointer (res, address, address ? g_object_unref : NULL);
-  g_simple_async_result_complete_in_idle (res);
+  res = g_task_new (G_OBJECT (self), NULL, callback, user_data);
+  g_task_return_pointer (res, address, address ? g_object_unref : NULL);
```
Comment (allisonkarlitskaya) [inline src/common/cockpitloopback.c]: minor nag: GTask already contains a `NULL` check on the destroy notify, so you don't need to conditionalise the callback

File: src/bridge/cockpitpolkitagent.c [hunk 2]
```diff
@@ -384,13 +381,9 @@ cockpit_polkit_agent_initiate_authentication_finish (PolkitAgentListener *listen
                                                      GAsyncResult *res,
                                                      GError **error)
 {
-  g_warn_if_fail (g_simple_async_result_is_valid (res, G_OBJECT (listener),
-                  cockpit_polkit_agent_initiate_authentication));
+  g_warn_if_fail (g_task_is_valid (res, G_OBJECT (listener)));
```
Comment (allisonkarlitskaya) [inline src/bridge/cockpitpolkitagent.c]: You can drop the `G_OBJECT()` cast.  GTask (wisely) takes `gpointer` here.

File: src/common/cockpitloopback.c [hunk 3]
```diff
@@ -84,12 +83,7 @@ cockpit_loopback_next_finish (GSocketAddressEnumerator *enumerator,
                               GAsyncResult *result,
                               GError **error)
 {
-  GSocketAddress *address;
```
Comment (allisonkarlitskaya) [inline src/common/cockpitloopback.c]: No `g_task_is_valid()` here?

File: src/bridge/cockpitpolkitagent.c (hunk 2 above)
Comment (martinpitt) [reply src/bridge/cockpitpolkitagent.c]: Nice! done.

File: src/common/cockpitloopback.c (hunk 1 above)
Comment (martinpitt) [reply src/common/cockpitloopback.c]: Convenient! Simplified.

File: src/common/cockpitloopback.c (hunk 3 above)
Comment (martinpitt) [reply src/common/cockpitloopback.c]: The original code doesn't have it, but indeed it seems cleaner to assert it. Added.

**Comment by martinpitt** [comment]
Bah, need to rebase to fix semaphore.

**Comment by martinpitt** [comment]
@Gundersanne: Does [this failure](https://209.132.184.41:8493/logs/pull-12868-20191002-201717-70527369-cockpit-project-cockpit--rhel-8-1-distropkg/log.html) ring a bell? It's not related to this change, it e. g. also occurs in [#12906](https://209.132.184.41:8493/logs/pull-12906-20191002-154040-09f077ed-cockpit-project-cockpit--rhel-8-1-distropkg/log.html#141-2). It's not a flake, it's consistently failing.

**Comment by martinpitt** [comment]
The [rhel-8-1-distropkg failure](https://209.132.184.41:8493/logs/pull-12868-20191002-201717-70527369-cockpit-project-cockpit--rhel-8-1-distropkg/log.html#112-2) is essentially what got fixed in PR #12806. The test failure was partially fixed in PR #12825, but this only applied to testFirewallPage, not to testNetworkingPage.

**Pull request merged**